### PR TITLE
BC-6 # `bm bmp pull` downloads config and code

### PR DESCRIPTION
**Pull request by jokeyrhyme**
- uses previously-designed approach for splitting code out of JSON files into separate files: https://github.com/blinkmobile/json-as-files.js

- just replaces files at the destination, no protection or warning so far

### CODE REVIEW

**Comment by jokeyrhyme** [comment]
Should be rebased and merged after BC-5 (#2) goes through.

**Comment by jokeyrhyme** [comment]
`git rebase`d, etc.

File: lib/whoami.js [hunk 1]
```diff
@@ -2,14 +2,31 @@
...
 // this module
 
 module.exports = {
 
+  getAuthentication (options) {
+    options = options || {};
+    const userConfigFile = values.userConfigFile(options.userConfigDir);
+    const origin = urlUtils.getOrigin(options.scope || '');
+    return jsonUtils.loadJsonObject(userConfigFile)
+      .catch(() => [])
```
Comment (simonmarklar) [inline lib/whoami.js]: I think this will swallow any errors thrown by `loadJsonObject`, would it be worth outputting a message to the console or log?

Comment (jokeyrhyme) [inline lib/whoami.js]: Fair enough. It'll either be a file-system error, or a JSON parsing error. It's probably good for debugging if these things are displayed to the user.

**Comment by jokeyrhyme** [comment]
@simonmarklar okay, I've stopped those functions from swallowing errors. I've done this in a slightly repetitive fashion, but I've deduped these functions in BC-7 (#5), so I didn't want to keep that future `git rebase` simple. :)

**Comment by simonmarklar** [comment]
cool, LGTM :+1:

**Pull request merged**